Trappist/src/Promact.Trappist.Repository/Questions/QuestionRepository.cs

```diff
@@ -31,34 +37,36 @@ public QuestionRepository(TrappistDbContext dbContext)
         /// </summary>
         /// <param name="singleMultipleAnswerQuestion"></param>
         /// <param name="singleMultipleAnswerQuestionOption"></param>
-        public async Task<QuestionAC>  AddSingleMultipleAnswerQuestionAsync(QuestionAC questionAC)
+        public async Task<QuestionAC> AddSingleMultipleAnswerQuestionAsync(QuestionAC questionAC)
         {
-            await _dbContext.SingleMultipleAnswerQuestion.AddAsync(questionAC.SingleMultipleAnswerQuestionAC.SingleMultipleAnswerQuestion);
-            foreach (SingleMultipleAnswerQuestionOption singleMultipleAnswerQuestionOptionElement in questionAC.SingleMultipleAnswerQuestionAC.SingleMultipleAnswerQuestionOption)
+            await _dbContext.SingleMultipleAnswerQuestion.AddAsync(questionAC.SingleMultipleAnswerQuestion.SingleMultipleAnswerQuestion);
+            foreach (SingleMultipleAnswerQuestionOption singleMultipleAnswerQuestionOptionElement in questionAC.SingleMultipleAnswerQuestion.SingleMultipleAnswerQuestionOption)
             {
-                //To-Do Change according to new model singleMultipleAnswerQuestionOptionElement.SingleMultipleAnswerQuestionID = singleMultipleAnswerQuestion.Id;
+                //To-Do Change according to new model
                 await _dbContext.SingleMultipleAnswerQuestionOption.AddAsync(singleMultipleAnswerQuestionOptionElement);
             }
             _dbContext.SaveChanges();
-            return(questionAC);
+            return (questionAC);
         }
-
+        
         /// <summary>
-        /// Add new code snippet question to the database
+        /// Adds New Code Snippet Question To The Database
         /// </summary>
         /// <param name="questionAC">Question data transfer object</param>
-        public async Task AddCodeSnippetQuestionAsync(QuestionAC questionAC)
+        public async Task AddCodeSnippetQuestionAsync(QuestionAC questionAC, ClaimsPrincipal User)
         {
-            var codeSnippetQuestion = Mapper.Map<CodeSnippetQuestionAC, CodeSnippetQuestion>(questionAC.CodeSnippetQuestionAC);
+            var codeSnippetQuestion = Mapper.Map<CodeSnippetQuestionAC, CodeSnippetQuestion>(questionAC.CodeSnippetQuestion);
+            var question = Mapper.Map<QuestionDetailAC, Question>(questionAC.Question);
+            question.ApplicationUser = _userManager.GetUserAsync(User).Result;
```

Comment:
Does it require entire user object to pass from controller to repository? I think the only email should be enough